cmd/device-definitions-api/populate_device_features.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"context"
+	"strings"
+
+	"github.com/DIMO-Network/device-definitions-api/internal/config"
+	"github.com/DIMO-Network/device-definitions-api/internal/infrastructure/db/models"
+	elastic "github.com/DIMO-Network/device-definitions-api/internal/infrastructure/elastic_search"
+	"github.com/DIMO-Network/shared/db"
+	"github.com/rs/zerolog"
+	"github.com/volatiletech/sqlboiler/v4/boil"
+)
+
+func prepareFeatureData(f map[string]map[string]int) []models.DeviceIntegrationFeatures {
+	ft := make([]models.DeviceIntegrationFeatures, len(f))
+
+	for k, v := range f {
+		feat := models.DeviceIntegrationFeatures{}
+		feat.ElasticProperty = k
+		feat.SupportLevel = 0
+
+		if v["doc_count"] > 0 {
+			feat.SupportLevel = 2
+		}
+
+		ft = append(ft, feat)
+	}
+
+	return ft
+}
+
+func populateDeviceFeaturesFromEs(ctx context.Context, logger zerolog.Logger, s *config.Settings) error {
+	pdb := db.NewDbConnectionFromSettings(ctx, &s.DB, true)
+	pdb.WaitForDB(logger)
+
+	es, _ := elastic.NewElasticSearch(s, logger)
+
+	resp, err := es.GetDeviceFeatures(s.Environment)
+	if err != nil {
+		return err
+	}
+
+	for _, f := range resp.Aggregations.Features.Buckets {
+		intID := strings.Split(f.Key, "/")[2]
+
+		for _, d := range f.DeviceDefinitions.Buckets {
+			ddID := d.Key
+
+			devices, err := models.DeviceIntegrations(
+				models.DeviceIntegrationWhere.DeviceDefinitionID.EQ(ddID),
+				models.DeviceIntegrationWhere.IntegrationID.EQ(intID),
+			).All(ctx, pdb.DBS().Reader)
+			// check if device exists
+			if err != nil {
+				logger.Error().Msgf("error occurred fetching device with integration id %s and deviceDefinitionId %s. error: %s", intID, ddID, err.Error())
+				continue
+			}
+
+			/* if len(devices) > 1 { // we have for multiple continents
+				// handle when we have region in elasticsearch
+			} */
+
+			if len(devices) < 1 {
+				// handle not found
+				logger.Error().Msgf("error could not find device with integration id %s and deviceDefinitionId %s", intID, ddID)
+				continue
+			}
+
+			feature := prepareFeatureData(d.Features.Buckets)
```

Comment:
In this JSON field it seems like we're putting
```
[{"elastic_key": "soc", "supportLevel": 2}, …]
```
If you want to use the Elastic key (I might have called it field, I think the Elastic people usually talk about fields) I think that's fine. In that case we can probably dump the `feature_key` column, then.

It would be nice to be consistent with casing, so writing `elasticKey` here.

What does support level 2 mean?

Reply:
Noted and corrected about the key consistency and also the primary key. The support levels for this task is 

0 - Not supported
1 - Maybe Supported
2 - Supported